backend/src/api.py

```diff
@@ -259,26 +261,72 @@ def contribute_review_container_logs(
 
 
 @app.get("/frontend/review/random")
-def frontend_review_random() -> FeedbackSchema:
-    if os.environ.get("ENV") == "production":
-        raise NotImplementedError("Reviewing is not ready yet")
-
+def frontend_review_random():
     random_feedback_file = Storator3000.get_random()
     with open(random_feedback_file) as random_file:
         content = json.loads(random_file.read())
-        return FeedbackSchema(**content)
+        return FeedbackSchema(**content).dict() | {"id": random_feedback_file.name.rstrip(".json")}
+
+
+def _get_text_from_feedback(item: dict) -> str:
+    if item["vote"] != 1:
+        return ""
 
+    return item["text"]
 
-@app.get("/frontend/review/latest")
-def frontend_review_latest() -> FeedbackSchema:
-    if os.environ.get("ENV") == "production":
-        raise NotImplementedError("Reviewing is not ready yet")
 
-    feedback_file = Storator3000.get_latest()
+def _parse_snippet(snippet: dict) -> dict:
+    return {
+        "start_index": snippet["start-index"],
+        "end_index": snippet["end-index"],
+        "user_comment": snippet["comment"],
+        "text": snippet["text"],
+    }
 
-    with open(feedback_file) as file:
-        content = json.loads(file.read())
-        return FeedbackSchema(**content)
+
+def _parse_logs(logs_orig: dict[str, FeedbackLogSchema], review_snippets: list[dict]) -> dict:
+    for name, item in logs_orig.items():
+        item.snippets = []
+        for snippet in review_snippets:
+            if snippet["file"] == name and snippet["vote"] == 1:
+                item.snippets.append(_parse_snippet(snippet))
+
+
+def _parse_feedback(review: dict, origin_id: int) -> FeedbackSchema:
+    original_file_path = find_file_by_name(f"{origin_id}.json", Path(FEEDBACK_DIR))
+    with open(original_file_path) as fp:
+        original_content = json.load(fp)
+        schema = FeedbackSchema(**original_content)
+        # no reason to store username
+        schema.username = None
+        schema.fail_reason = _get_text_from_feedback(review["fail_reason"])
+        schema.how_to_fix = _get_text_from_feedback(review["how_to_fix"])
+        _parse_logs(schema.logs, review["snippets"])
+        return schema.dict(exclude_unset=True)
+
+
+@app.post("/frontend/review")
+async def store_random_review(feedback_input: Request) -> OkResponse:
+    # TODO: temporary silly solution until database is created
+    #  (missing provider but we can dig it from original feedback file)
+    reviews_dir = Path(REVIEWS_DIR)
+    parsed_reviews_dir = reviews_dir / "parsed"
+    parsed_reviews_dir.mkdir(parents=True, exist_ok=True)
+    content = await feedback_input.json()
+    original_file_id = content.pop("id")
+    # avoid duplicates - same ID can be reviewed multiple times
+    file_name = original_file_id + "-" + str(int(datetime.now().timestamp()))
```

Comment:
I guess this will be slightly better readable a f-string or using format() method.